Source/Core/Core/PowerPC/Interpreter/Interpreter_FPUtils.h

```diff
@@ -287,12 +287,143 @@ inline FPResult NI_sub(PowerPC::PowerPCState& ppc_state, double a, double b)
+  // it's not particularly difficult to make your own examples as well,
+  // but of course these happening in practice is going to be absurdly uncommon most of the time.
+  //
+  // Currently Dolphin supports:
+  // - Correct ordering of NaN checking (for both double and single precision)
+  // - Rounding frC up
+  // - Rounding only once for single precision inputs (this will be the large majority of cases!)
+  //   - Currently this is interpreter-only.
+  //     This can be implemented in the JIT just as easily, though.
+  //     Eventually the JITs should hopefully support detecting back to back
+  //     single-precision operations, which will lead to no overhead at all.
+  //
+  // Currently it does not support:
+  // - Handling frC overflowing to an unreachable value
+  //   - This is simple enough to check for and handle properly, but the likelihood of it occuring
+  //     is so low that it's not worth it to check for it for the rare accuracy improvement.
+  // - Rounding only once for inputs with single precision mantissas but double precision exponents
+  //   - This one is also very simple again is not really something that would happen.
+  //     It's also the most likely one to occur, as paired single move operations similarly only
+  //     round the mantissa, not the exponent, and there are games which which do in fact
+  //     utilize this (for example, any games which use nw4r -- this includes Mario Kart Wii,
+  //     although no ghosts are known which desync on Dolphin because of this or even the
+  //     single -> double -> single precision FMA issue)
+  // - Rounding only once for inputs with double precision mantissas
+  //
+  // All of these can be resolved in a software float emulation method, or by using things such as
+  // error-free float algorithms, but the nature of both of these lead to incredible speed costs,
+  // and with the extreme rarity of anything beyond what's currently handled mattering, the other
+  // cases don't seem to have any reason to be implemented as of now.
+
+  FPResult result;
+
+  // In double precision, just doing the normal operation will be exact with no issues.
+  if (!single)
+    result.value = std::fma(a, c, sub ? -b : b);
+  else
+  {
+    // For the single precision case, all we currently do is rounding frC properly,
+    // then check if the single precision case will work,
+    // and if it does, we perform a single precision fma instead.
+    const double c_round = Force25Bit(c);
+
+    const float a_float = static_cast<float>(a);
+    const float b_float = static_cast<float>(b);
+    const float c_float = static_cast<float>(c_round);
+
+    if (static_cast<double>(a_float) == a && static_cast<double>(b_float) == b &&
+        static_cast<double>(c_float) == c_round)
+      result.value = static_cast<double>(std::fma(a_float, c_float, sub ? -b_float : b_float));
+    else
+      result.value = std::fma(a, c_round, sub ? -b : b);
```

Comment:
```suggestion
    {
      result.value = static_cast<double>(std::fma(a_float, c_float, sub ? -b_float : b_float));
    }
    else
    {
      result.value = std::fma(a, c_round, sub ? -b : b);
    }
```

(Due to the condition being longer than one line)